File: go/problems/83_remove_duplicates_from_sorted_list.go
package main

func deleteDuplicates(head *ListNode) *ListNode {
	if head == nil || head.Next == nil {
		return head
	}

	temp := head
	for temp.Next != nil {
		if temp.Val == temp.Next.Val {
			temp.Next = temp.Next.Next
			continue
		}
		temp = temp.Next
	}

	return head
}

File: ts/problems/83-remove-duplicates-from-sorted-list.ts
export type ListNode = {
  val: number;
  next: ListNode | null;
};

export function deleteDuplicates(head: ListNode | null): ListNode | null {
  if (head === null || head.next === null) {
    return head;
  }

  let current = head;

  while (current.next !== null) {
    if (current.val === current.next.val) {
      current.next = current.next.next;
      continue;
    }
    current = current.next;
  }

  return head;
}
